axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/invoice/generator/InvoiceLineGeneratorSupplyChain.java

```diff
@@ -117,14 +117,17 @@ protected InvoiceLineGeneratorSupplyChain(
       this.taxLine = saleOrderLine.getTaxLine();
       this.discountTypeSelect = saleOrderLine.getDiscountTypeSelect();
       this.isTitleLine = saleOrderLine.getTypeSelect() == SaleOrderLineRepository.TYPE_PACK;
+      this.unit = saleOrderLine.getUnit();
+      this.qty = saleOrderLine.getQty();
     } else if (purchaseOrderLine != null) {
       this.isTitleLine = purchaseOrderLine.getIsTitleLine();
-      this.purchaseOrderLine = purchaseOrderLine;
       this.discountAmount = purchaseOrderLine.getDiscountAmount();
       this.price = purchaseOrderLine.getPrice();
       this.priceDiscounted = purchaseOrderLine.getPriceDiscounted();
       this.taxLine = purchaseOrderLine.getTaxLine();
       this.discountTypeSelect = purchaseOrderLine.getDiscountTypeSelect();
+      this.unit = purchaseOrderLine.getUnit();
+      this.qty = purchaseOrderLine.getQty();
```

Comment:
Qty should not get from saleOrderLine or purchaseOrderLine but depends of stockMoveLine. We can have partial stock move.